mapflow/dialogs/mosaic_dialog.py

```diff
@@ -0,0 +1,59 @@
+from PyQt5 import uic
+from PyQt5.QtWidgets import QWidget, QDialogButtonBox
+
+from ..schema.data_catalog import MosaicCreateSchema, MosaicUpdateSchema, MosaicReturnSchema
+from .dialogs import ui_path, plugin_icon
+
+class MosaicDialog(*uic.loadUiType(ui_path/'mosaic_dialog.ui')):
+    def __init__(self, parent: QWidget) -> None:
+        """A dialog for adding or editing mosaics"""
+        super().__init__(parent)
+        self.setupUi(self)
+        self.setWindowIcon(plugin_icon)
+        self.ok = self.buttonBox.button(QDialogButtonBox.Ok)
+
+        self.mosaicName.textChanged.connect(self.on_name_change)
+
+    def on_name_change(self):
+        if not self.mosaicName.text():
+            self.ok.setEnabled(False)
+            self.ok.setToolTip(self.tr("Mosaic name must not be empty!"))
+        else:
+            self.ok.setEnabled(True)
+            self.ok.setToolTip("")
+
+class CreateMosaicDialog(MosaicDialog):
+    def __init__(self, parent: QWidget):
+        super().__init__(parent)
+        self.ok.setEnabled(False)
+
+    def setup(self):
+        self.setWindowTitle("")
+        self.mosaicName.setText("")
+        self.mosaicTags.setText("")
+        self.exec()
+
+    def mosaic(self):
+        if not self.mosaicName:
+            raise AssertionError("Mosaic name must not be empty!")
+        return MosaicCreateSchema(name = self.mosaicName.text(),
+                                  tags = self.mosaicTags.text() or None)
+
+class UpdateMosaicDialog(MosaicDialog):
+    def __init__(self, parent: QWidget):
+        super().__init__(parent)
+        self.ok.setEnabled(True)
+
+    def setup(self, mosaic: MosaicReturnSchema):
+        if not mosaic:
+            raise TypeError("UpdateMosaicDialog requires a mosaic to update")
+        self.setWindowTitle(self.tr("Edit mosaic {}").format(mosaic.name))
+        self.mosaicName.setText(mosaic.name)
+        self.mosaicTags.setText(mosaic.tags or "")
+        self.exec()
+
+    def mosaic(self):
+        if not self.mosaicName:
+            raise AssertionError("Mosaic name must not be empty!")
+        return MosaicUpdateSchema(name = self.mosaicName.text(),
+                                  detags = self.mosaicTags.text() or None)
```

Comment:
Here MosaicUpdateSchema awaits `tags` keyword arguments, not `detags`

To debug such errors, search for the strange words from the error message in your code!

Reply:
Oops, missed this, thanks, but there is a bigger problem now:
`curl 'https://whitemaps-duty.mapflow.ai/rest/rasters/mosaic/2626a27b-efc2-4d00-a1cc-17fb40170846' -H 'Content-Type: application/json' -H 'x-plugin-version: 2.6.0' -H 'authorization: Basic ZC5pZ25hdGVua29AZ2VvYWxlcnQuaW86T3RCall4MWFkRFNCNzZLdFphZGU=' -H 'User-Agent: Mozilla/5.0 QGIS/32800/Windows 10 Version 2009'  --data '{"name": "updated_mosaic", "tags": "some, updated, tags"}' --compressed`
results in:
`{"detail":"Method Not Allowed"}`
Any ideas of what can be wrong?

Reply:
Here is a JSON from debugging tools:
{
  "Bytes Received": 95,
  "Bytes Total": 95,
  "Cache (control)": "Always load from network, do not check cache",
  "Cache (save)": "Result cannot be stored in cache",
  "Content": {
    "Data": "{\"name\": \"updated_mosaic\", \"tags\": \"some, updated, tags\"}"
  },
  "Headers": {
    "Content-Type": "application/json",
    "User-Agent": "Mozilla/5.0 QGIS/32800/Windows 10 Version 2009",
    "authorization": "Basic ZC5pZ25hdGVua29AZ2VvYWxlcnQuaW86T3RCall4MWFkRFNCNzZLdFphZGU=",
    "x-plugin-version": "2.6.0"
  },
  "Initiator": "unknown",
  "Operation": "PUT",
  "Replies": 2,
  "Reply": {
    "Cache (result)": "Read from network",
    "Error": "Error transferring https://whitemaps-duty.mapflow.ai/rest/rasters/mosaic/2626a27b-efc2-4d00-a1cc-17fb40170846 - server replied: Unprocessable Entity",
    "Error Code": "299",
    "Headers": {
      "Content-Length": "95",
      "Content-Type": "application/json",
      "Date": "Fri, 26 Jul 2024 08:52:14 GMT",
      "Strict-Transport-Security": "max-age=15724800; includeSubDomains"
    },
    "Status": "422"
  },
  "Thread": "0x0000019a88f68c50",
  "Total time (ms)": 280,
  "URL": "https://whitemaps-duty.mapflow.ai/rest/rasters/mosaic/2626a27b-efc2-4d00-a1cc-17fb40170846"
}